tests/posix/pthread_rwlock/src/posix_rwlock.c

```diff
@@ -81,7 +81,10 @@ static void test_rw_lock(void)
 	/* Creating N premptive threads in increasing order of priority */
 	for (i = 0; i < N_THR; i++) {
 		pthread_attr_destroy(&attr[i]);
-		pthread_attr_init(&attr[i]);
+		if (pthread_attr_init(&attr[i]) != 0) {
+			zassert_false(pthread_attr_init(&attr[i]),
```

Comment:
Shouldn't the zassert_equal() call be made only in the first call to pthread_attr_init()? Trying again shouldn't make a difference that's going to make the assertion fail

Reply:
yeah we can do it that way.  I just used the method others were using to do this in other tests.